lib/DependencyScan/ScanDependencies.cpp

```diff
@@ -1242,14 +1242,21 @@ bool swift::dependencies::scanDependencies(CompilerInstance &instance) {
   GlobalModuleDependenciesCache globalCache;
   if (opts.ReuseDependencyScannerCache)
     deserializeDependencyCache(instance, globalCache);
-
-  auto ModuleCachePath = getModuleCachePathFromClang(
-               Context.getClangModuleLoader()->getClangInstance());
-
+  // Wrap the filesystem with a caching `DependencyScanningWorkerFilesystem`
+  auto realFS = instance.getSourceMgr().getFileSystem();
+  llvm::IntrusiveRefCntPtr<
+      clang::tooling::dependencies::DependencyScanningWorkerFilesystem>
+      depFS =
+          new clang::tooling::dependencies::DependencyScanningWorkerFilesystem(
+              globalCache.getSharedFilesystemCache(), realFS);
+  instance.getSourceMgr().setFileSystem(depFS);
```

Comment:
Nit: can we extract this logic to a common utility taking `instance` as a parameter?